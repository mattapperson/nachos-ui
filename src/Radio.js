import React from 'react'
import PropTypes from 'prop-types'
import Checkbox from './Checkbox'

const Radio = (props) => {
  return <Checkbox {...props} />
}

Radio.propTypes = {
  checkType: PropTypes.string,
  kind: PropTypes.string,
}

Radio.defaultProps = {
  checkType: 'circle',
  kind: 'circle',
}

export default Radio
