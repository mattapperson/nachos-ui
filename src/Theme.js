import React, { Component } from "react";
import PropTypes from "prop-types";
import { StyleSheet } from "react-native";
import defaultBranding from "./branding_defaults";

export const ThemeContext = React.createContext({
  theme: {}
});

export class Provider extends React.Component {
  state = {};
  static getDerivedStateFromProps(nextProps, prevState) {
    if (JSON.stringify(nextProps.theme) !== JSON.stringify(prevState)) {
      //return this.createTheme(nextProps.theme);
    }
    return null;
  }

  themeConfigs = {};
  branding = defaultBranding;

  updateThemeConfig = (name, themeConfig) => {
    themeConfig = JSON.parse(
      JSON.stringify(themeConfig).replace(/@([\w_-]+)/gm, (match, key) => {
        if (!this.branding[key]) console.log(key);

        return this.branding[key];
      })
    );

    // if (name === "H1") {
    //   console.log(themeConfig);
    // }
    return (this.themeConfigs[name] = {
      computedStyle: themeConfig.style
        ? StyleSheet.create(themeConfig.style)
        : {},
      props: themeConfig.settings
    });
  };

  render() {
    return (
      <ThemeContext.Provider
        value={{
          theme: this.themeConfigs,
          updateThemeConfig: this.updateThemeConfig
        }}
      >
        {this.props.children}
      </ThemeContext.Provider>
    );
  }
}

export function withTheme(componentName, ThemedComponent) {
  return class extends React.Component {
    constructor() {
      super();
      try {
        ThemeContext._currentValue.updateThemeConfig(
          componentName,
          ThemedComponent.themeConfig
        );
      } catch (e) {
        console.warn(`failed to update theme for ${componentName}`, e);
      }
    }

    isObject(item) {
      return item && typeof item === "object" && !Array.isArray(item);
    }

    mergeDeep(target, ...sources) {
      if (!sources.length) return target;
      const source = sources.shift();

      if (this.isObject(target) && this.isObject(source)) {
        for (const key in source) {
          if (this.isObject(source[key])) {
            if (!target[key]) Object.assign(target, { [key]: {} });
            this.mergeDeep(target[key], source[key]);
          } else {
            Object.assign(target, { [key]: source[key] });
          }
        }
      }

      return this.mergeDeep(target, ...sources);
    }
    render() {
      return (
        <ThemeContext.Consumer>
          {({ theme }) => {
            const computedTheme =
              (theme[componentName] || {}).computedStyle || {};

            const fullTheme = this.props.theme
              ? this.mergeDeep({}, computedTheme, this.props.theme)
              : computedTheme;

            return (
              <ThemedComponent
                {...(theme[componentName] || {}).props || {}}
                {...this.props}
                theme={fullTheme}
              />
            );
          }}
        </ThemeContext.Consumer>
      );
    }
  };
}
