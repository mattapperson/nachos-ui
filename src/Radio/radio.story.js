import { storiesOf } from "../storybook";
import { withInfo } from "../storybook";
import React from "react";
import Radio from "./index";
import Switcher from "../Switcher";

import { View } from "react-native";

const Example = () => {
  const radioStyle = { margin: 10 };

  return (
    <View style={{ flex: 1, flexDirection: "column" }}>
      <Switcher defaultSelected="bus">
        <Radio value="car" style={radioStyle} />
        <Radio value="bus" style={radioStyle} />
      </Switcher>
    </View>
  );
};

storiesOf("Radio", module).add("default", withInfo()(Example));

export default Example;
