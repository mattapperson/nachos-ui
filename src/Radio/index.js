import React from "react";
import PropTypes from "prop-types";
import Checkbox from "../Checkbox";
import { withTheme } from "../Theme";
import { View } from "react-native";

const Radio = ({ theme, ...rest }) => {
  return (
    <Checkbox
      checkComponent={<View style={[theme.check]} />}
      theme={theme}
      {...rest}
    />
  );
};

Radio.themeConfig = {
  props: {
    iconName: "md-checkmark",
    iconColor: "#fff",
    iconSize: 18
  },
  style: {
    base: {
      position: "relative",
      width: 22,
      height: 22,
      borderWidth: 2,
      justifyContent: "center",
      alignItems: "center",
      borderColor: "#ececee",
      borderRadius: 9999
    },
    active: {
      backgroundColor: "#fff"
    },
    check: {
      width: 12,
      height: 12,
      borderRadius: 6,
      backgroundColor: "@primaryColor"
    }
  }
};

Radio.propTypes = {
  checkType: PropTypes.string,
  kind: PropTypes.string
};

Radio.defaultProps = {
  checkType: "circle",
  kind: "circle"
};

export default withTheme("Radio", Radio);
