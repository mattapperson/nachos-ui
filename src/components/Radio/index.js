import React from "react";
import PropTypes from "prop-types";
import Checkbox from "../Checkbox";
import { withTheme } from "../Theme";
import { View } from "react-native";

const Radio = ({ theme, ...rest }) => {
  return (
    <Checkbox
      checkComponent={<View style={[theme.check]} />}
      theme={theme}
      {...rest}
    />
  );
};

Radio.themeConfig = {
  props: {},
  style: {
    base: {
      position: "relative",
      width: 22,
      height: 22,
      borderWidth: 2,
      justifyContent: "center",
      alignItems: "center",
      borderColor: "#ececee",
      borderRadius: 9999
    },
    active: {
      backgroundColor: "#fff"
    },
    check: {
      width: 12,
      height: 12,
      borderRadius: 6,
      backgroundColor: "@primaryColor"
    }
  }
};

Radio.propTypes = {
  selected: PropTypes.bool,
  disabled: PropTypes.bool,
  onValueChange: PropTypes.func,
  value: PropTypes.string
};

Radio.defaultProps = {
  iconSize: 18,
  activeOpacity: 0.8,
  disabled: false,
  disabledStyle: { opacity: 0.3 },
  selected: false,
  onValueChange: () => {}
};

export default withTheme("Radio", Radio);
