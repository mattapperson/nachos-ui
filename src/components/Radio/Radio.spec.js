import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "../Theme";
import Radio from "../Radio";

it("should render correctly", () => {
  const component = renderer.create(
    <Provider>
      <Radio />{" "}
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
