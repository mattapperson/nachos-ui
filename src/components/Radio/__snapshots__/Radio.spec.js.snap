// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <div
    className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <div
      className="rn-alignItems-1awozwy rn-borderTopColor-rv39ua rn-borderRightColor-19i1o9h rn-borderBottomColor-to9v4f rn-borderLeftColor-uaesm1 rn-borderTopLeftRadius-1bxrh7q rn-borderTopRightRadius-waaub4 rn-borderBottomRightRadius-sqtsar rn-borderBottomLeftRadius-qb5c1y rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-1jxfwug rn-borderRightWidth-18p6if4 rn-borderBottomWidth-wgabs5 rn-borderLeftWidth-dwliz8 rn-boxSizing-deolkf rn-cursor-1loqt21 rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-18yzcnr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-touchAction-1otgn73 rn-transitionDuration-eafdt9 rn-transitionProperty-1i6wzkk rn-userSelect-lrvibr rn-width-yc9v9c rn-zIndex-1lgpqti"
      data-focusable={true}
      disabled={false}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      tabIndex="0"
    />
  </div>,
  " ",
]
`;
